lib/api/repository/auth_repository.dart

```diff
@@ -0,0 +1,32 @@
+import 'package:injectable/injectable.dart';
+import 'package:survey/api/grant_type.dart';
+import 'package:survey/api/request/login_request.dart';
+import 'package:survey/api/response/login_response.dart';
+import 'package:survey/api/service/auth_service.dart';
+import 'package:survey/env_variables.dart';
+
+abstract class AuthRepository {
+  Future<LoginResponse> login(
+      {required String email, required String password});
+}
+
+@Singleton(as: AuthRepository)
+class AuthRepositoryImpl extends AuthRepository {
+  final AuthService _authService;
+
+  AuthRepositoryImpl(this._authService);
+
+  @override
+  Future<LoginResponse> login(
+      {required String email, required String password}) {
+    return _authService.login(
+      LoginRequest(
+        grantType: GrantType.password.value,
+        email: email,
+        password: password,
+        clientId: EnvVariables.clientId,
+        clientSecret: EnvVariables.clientSecret,
+      ),
+    );
+  }
```

Comment:
Should the Repository layer return model instead of response model back to Usecase layer?